actionpack/lib/action_dispatch/journey/formatter.rb

```diff
@@ -185,6 +185,17 @@ def missing_keys(route, parts)
           missing_keys
         end
 
+        def valid_key?(test, value)
+          test.match?(value) ||
+          test.match?(escape_value(value))
```

Comment:
I think it's not a good sign that we have to support both a non-escaped and an escaped value.

The fact that it doesn't work only with wildcards is I think related to the constraint that is added to wildcards, and the fact that only causes an issue with newline is I think related to the fact that constraints are "anchored", which prevents newlines from appearing inside the value:
https://github.com/rails/rails/blob/4a0da52083eee72c14b5015b6ce30018ee3b95da/actionpack/lib/action_dispatch/journey/path/pattern.rb#L188

I don't think that's the right regexp actually, that's the one I'm seeing with our test failure which is actually related to a dynamic segment… The thing is that when people use constraints, they probably mean the value literally, so you want to check unescaped, but when a wildcard constraint is set, you either want not to anchor it to allow for newlines, or to escape the value for the newline to not take effect. I think we should look at changing the constraint set by a wildcard segment.